dm/pkg/dumpling/utils.go

```diff
@@ -40,12 +42,30 @@ var DumplingDefaultTableFilter = []string{"*.*", export.DefaultTableFilter}
 // since v2.0.0, dumpling maybe configured to output master status after connection pool is established,
 // we return this location as well.
 func ParseMetaData(filename, flavor string) (*binlog.Location, *binlog.Location, error) {
-	invalidErr := fmt.Errorf("file %s invalid format", filename)
 	fd, err := os.Open(filename)
 	if err != nil {
 		return nil, nil, err
 	}
 	defer fd.Close()
+	return parseMetaDataByReader(filename, flavor, fd)
+}
+
+// ParseMetaDataByExternalStore parses mydumper's output meta file by `ExternalStorage` which supports s3 and returns binlog location.
+// since v2.0.0, dumpling maybe configured to output master status after connection pool is established,
+// we return this location as well.
+func ParseMetaDataByExternalStore(ctx context.Context, dir, filename, flavor string) (*binlog.Location, *binlog.Location, error) {
```

Comment:
why not enchance `ParseMetaData` using `expackage`?

Reply:
ok, the same as `getMydumpMetadataByExternalStorage`.